### DISCUSSION TITLE
Is it possible to build libusb only with POSIX headers?

### DISCUSSION THREAD

**Discussion by OctopusET** [libusb/libusb#1523] [Q&A] [answered]
I'm trying to build libusb with [cosmopolitan](https://github.com/jart/cosmopolitan) to run libusb everywhere.

When I build libusb with cosmopolitan, I get this error `Linux netlink header not found`. So I wonder, is it possible to build `libusb` without netlink header and only with POSIX headers?

**Comment by mcuee** [comment, marked as answer]
I believe it is not possible to get what you want. libusb uses platform specific codes.

Under Linux you need either libudev (preferred) or Netlink (fallback).

**Comment by seanm** [comment]
>So I wonder, is it possible to build libusb without netlink header and only with POSIX headers?

We could turn the question around: does POSIX specify any way to communicate with USB devices?  AFAIK, the answer is no. And so libusb has no choice but to have a bunch of OS-specific cases.
